8339247: Create release notes for JavaFX 23

kevinrushforth opened the pull request:
This PR adds the release notes for the JavaFX 23 release. This will first go into `master`, and then be backported to the `jfx23` branch so it will be available in that branch when JavaFX 23 is published (and from there also synced into the `jfx23u` repo).

The following filter was used to produce the list of issues fixed in JavaFX 23:

https://bugs.openjdk.org/issues/?filter=45422

Additionally, we had three issues with a `release-note=yes` label, which are included in the list of important changes.

I will wait until closer to the September 17th release date to integrate this.


A [suggestion was made last time](https://github.com/openjdk/jfx/pull/1402#issuecomment-1998821014) to split out the new features (something that adds new functionality to the platform) from enhancements. If we decide to do that, here is my proposed split:

## List of New Features

Issue key|Summary|Subcomponent
---------|-------|------------
[JDK-8092102](https://bugs.openjdk.org/browse/JDK-8092102)|Labeled: textTruncated property|controls
[JDK-8313138](https://bugs.openjdk.org/browse/JDK-8313138)|Scrollbar Keyboard enhancement|controls
[JDK-8267565](https://bugs.openjdk.org/browse/JDK-8267565)|Support "@3x" and greater high-density image naming convention|graphics
[JDK-8311895](https://bugs.openjdk.org/browse/JDK-8311895)|CSS Transitions|graphics
[JDK-8282999](https://bugs.openjdk.org/browse/JDK-8282999)|Add support for EXT-X-MEDIA tag in HTTP Live Streaming|media

## List of Other Enhancements

Issue key|Summary|Subcomponent
---------|-------|------------
[JDK-8321603](https://bugs.openjdk.org/browse/JDK-8321603)|Bump minimum JDK version for JavaFX to JDK 21|build
[JDK-8322748](https://bugs.openjdk.org/browse/JDK-8322748)|Caret blinking in JavaFX should only stop when caret moves|controls
[JDK-8322964](https://bugs.openjdk.org/browse/JDK-8322964)|Optimize performance of CSS selector matching|graphics
[JDK-8324182](https://bugs.openjdk.org/browse/JDK-8324182)|Deprecate for removal SimpleSelector and CompoundSelector classes|graphics
[JDK-8325900](https://bugs.openjdk.org/browse/JDK-8325900)|Emit a warning on macOS if AWT has set the NSAppearance|graphics

<!-- Anything below this marker will be automatically updated, please do not edit manually! -->
---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8339247](https://bugs.openjdk.org/browse/JDK-8339247): Create release notes for JavaFX 23 (**Task** - P2)(⚠️ The fixVersion in this issue is [jfx23] but the fixVersion in .jcheck/conf is jfx24, a new backport will be created when this pr is integrated.)


### Reviewers
 * [Andy Goryachev](https://openjdk.org/census#angorya) (@andy-goryachev-oracle - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jfx.git pull/1553/head:pull/1553` \
`$ git checkout pull/1553`

Update a local copy of the PR: \
`$ git checkout pull/1553` \
`$ git pull https://git.openjdk.org/jfx.git pull/1553/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 1553`

View PR using the GUI difftool: \
`$ git pr show -t 1553`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jfx/pull/1553.diff">https://git.openjdk.org/jfx/pull/1553.diff</a>

</details>


### Webrev
[Link to Webrev Comment](https://git.openjdk.org/jfx/pull/1553#issuecomment-2321149845)

kevinrushforth commented:
Reviewers @johanvos @abhinayagarwal

kevinrushforth commented:
> A https://github.com/openjdk/jfx/pull/1402#issuecomment-1998821014 to split out the new features (something that adds new functionality to the platform) from enhancements.

@andy-goryachev-oracle @nlisker @johanvos @abhinayagarwal Do any of you have an opinion on this?

nlisker commented:
There are https://openjfx.io/highlights/22/, https://openjfx.io/highlights/21/ etc. I think they do a good job at separating APIs from improvements, so whatever the highlights for 23 will say needs to be aligned with the ones here, in my opinion.

The current separation looks fine.

andy-goryachev-oracle reviewed (approved):
that's a lot of bug fixes!
all that I wanted to be listed are listed.  thank you.

abhinayagarwal commented:
The current separation looks good to me too.

I have a question related to issues to be included: we tend to normally not include backports in release notes. However, issues like [JDK-8299738](https://bugs.openjdk.org/browse/JDK-8299738) and [JDK-8335630](https://bugs.openjdk.org/browse/JDK-8335630) are included. Are these included because the backports were successfully made before the release of `jfx23`?

kevinrushforth commented:
> The current separation looks good to me too.

A question then for both you and @nlisker :

Should I split the "Enhancements" section into two groups as shown in the PR Description, or leave it as a single group?

> I have a question related to issues to be included: we tend to normally not include backports in release notes. However, issues like [JDK-8299738](https://bugs.openjdk.org/browse/JDK-8299738) and [JDK-8335630](https://bugs.openjdk.org/browse/JDK-8335630) are included. Are these included because the backports were successfully made before the release of `jfx23`?

We started including backports as of JavaFX 21 because of the change to integrate fixes first into mainline and use backport-style PRs into the stabilization branch. See [this PR comment](https://github.com/openjdk/jfx/pull/1239#discussion_r1323664831) in the JavaFX 21 release notes PR.

nlisker commented:
> Should I split the "Enhancements" section into two groups as shown in the PR Description, or leave it as a single group?

@abhinayagarwal Do you write the highlights page https://openjfx.io/highlights/22/? Will the same split be done there?

I think the split should be kept.

andy-goryachev-oracle commented:
Just noticed: we might want to title case "Issue key" in both tables' headers.

kevinrushforth commented:
> > Should I split the "Enhancements" section into two groups as shown in the PR Description, or leave it as a single group?
...
> I think the split should be kept.

Done.

> Just noticed: we might want to title case "Issue key" in both tables' headers.

Fixed.

andy-goryachev-oracle commented:
Just a suggestion - sharing some tricks of writing the .md format:

It might be easier to review if the pipe characters in the table were aligned, and have at least one space separation from the previous cell's text.  This makes it easier to read in a code editor and does not produce spelling errors when the raw text is pasted in Word.

It adds some work though.  

Example:

```
Issue Key                                                  |Summary                         |Subcomponent
-----------------------------------------------------------|--------------------------------|------------
[JDK-8092102](https://bugs.openjdk.org/browse/JDK-8092102) |Labeled: textTruncated property |controls
[JDK-8313138](https://bugs.openjdk.org/browse/JDK-8313138) |Scrollbar Keyboard enhancement  |controls
```

nlisker commented:
I forgot I noted for myself some docs typos for this version. Obviously not critical to get them into 23, but if there's time before this is merged (should be until September 9th), I can create a PR quickly along the usual lines that we did in previous versions.

kevinrushforth commented:
> I forgot I noted for myself some docs typos for this version. Obviously not critical to get them into 23, but if there's time before this is merged (should be until September 9th), I can create a PR quickly along the usual lines that we did in previous versions.

We are already past the deadline for 23 (it was last Thursday), so they can go into 24.

kevinrushforth commented:
> Just a suggestion - sharing some tricks of writing the .md format:
> 
> It might be easier to review if the pipe characters in the table were aligned, and have at least one space separation from the previous cell's text. This makes it easier to read in a code editor and does not produce spelling errors when the raw text is pasted in Word.

I'll consider this for next time if not too much extra work.

kevinrushforth commented:
I plan to integrate this early tomorrow morning (Sep 12th). I will then backport it to the `jfx23` branch (I will _not_ trigger a new build of `jfx23` as a result).

kevinrushforth commented:
/integrate
